src/libs/ReportActionsUtils.ts

```diff
@@ -631,6 +661,126 @@ function isNotifiableReportAction(reportAction: OnyxEntry<ReportAction>): boolea
     return actions.includes(reportAction.actionName);
 }
 
+function getChannelLogMemberAction(reportAction: OnyxEntry<ReportAction>): MessageActionItemChanelLog[] {
+    const messageItems: MessageActionItemChanelLog[] = [];
+    const isInviteRoom = isInvitedRoom(reportAction);
+
+    const verb = isInviteRoom ? Localize.translateLocal('workspace.invite.invited') : Localize.translateLocal('workspace.invite.removed');
+
+    messageItems.push({
+        kind: 'text',
+        content: `${verb} `,
+    });
+
+    const originalMessage = reportAction?.originalMessage as CommonOriginalMessage;
+    const targetAccountIDs: number[] = originalMessage?.targetAccountIDs ?? [];
+    const personalDetails = PersonalDetailsUtils.getPersonalDetailsByIDs(targetAccountIDs, 0);
+
+    const mentions = targetAccountIDs.map((accountID) => {
+        const personalDetail = personalDetails.find((personal) => personal.accountID === accountID);
+
+        if (personalDetail) {
+            const displayNameOrLogin =
+                LocalePhoneNumber.formatPhoneNumber(personalDetail.login ?? '') || (personalDetail?.displayName ? personalDetail?.displayName : Localize.translateLocal('common.hidden'));
+            return {content: `@${displayNameOrLogin}`, accountID};
+        }
+        return {content: '', accountID: 0};
```

Comment:
On the one hand, I understand the defense-in-depth approach. "Better safe than sorry".

On the other hand, I don't like using `0` as `accountID`, I have no idea whether `0` will actually work here, and I'm not excited to test this case.

I would suggest adding a `mapNotNull` util in a new file like `ArrayUtils.ts`.

```ts
function mapNotNull<T, U>(array: T[], transform: (value: T) => U | null): ReadonlyArray<U> {
    const result: U[] = [];
    for (const item of array) {
        const transformed = transform(item);
        if (transformed !== null) {
            result.push(transformed);
        }
    }
    return result;
}
```

Then we can just easily skip the "invalid" `personalDetail` (which is quite unlikely, but technically possible) and just not generate a mention item for it.

Reply:
With a `for` loop?

Reply:
You can do this the way you feel is more straightforward. If I see something worth commenting on, I'll comment on then next revision.

Reply:
> I think we need to keep this logic, and update to display `Hidden` if personalDetail is not found in onyx. Because when logout and login again `MentionUserRenderer` will display `Hidden`.

I'm fine with this